### PR TITLE
Adding support at the frontend to get clustermap state

### PR DESCRIPTION
**Pull request by vgkholla**
Followup on #1065 to get clustermap state via HTTP request

### CODE REVIEW

File: ambry-api/src/main/java/com.github.ambry/clustermap/DiskId.java [hunk 1]
```diff
@@ -39,4 +42,9 @@
    * @return the raw capacity in bytes
    */
   long getRawCapacityInBytes();
+
+  /**
+   * @return a snapshot which includes information that the implementation considers relevant.
+   */
+  JSONObject getSnapshot();
```
Comment (lightningrob) [inline ambry-api/src/main/java/com.github.ambry/clustermap/DiskId.java]: Why not add this to the Resource superinterface?  Does any subinterface not need this?
Even if not every class needs it, we can add an intermediate interface like SnapshottableResource.

**Comment by lightningrob** [comment]
This looks very useful.  Who will have access to query the cluster map?  Will it be all accounts similar to HEAD access or more limited?

**Comment by vgkholla** [comment]
If an `AclService` implementation is available, then a custom ACL can be defined to handle this (similar to `GetPeers` and `GetReplica`)

File: ambry-api/src/main/java/com.github.ambry/clustermap/DiskId.java (hunk 1 above)
Comment (vgkholla) [reply ambry-api/src/main/java/com.github.ambry/clustermap/DiskId.java]: Makes sense - will look at this

Comment (vgkholla) [reply ambry-api/src/main/java/com.github.ambry/clustermap/DiskId.java]: Did this in #1065

**Comment by vgkholla** [comment]
@lightningrob @cgtz made changes after the other PR was merged and added a test for `GetClusterMapSnapshotHandler`. PTAL

**Pull request merged**